concurrent_server.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import server
+
+
+def set_server(conn, adr):
+    while True:
```

Comment:
This outer loop is useless.  When using a tool like `gevent` or `tornado` to build a server process, that tool provides for you the outer server loop, which is responsible for listening to the server socket and accepting new connection requests.  

This part here that you are writing is intended to handle one and only one connection.  It is invoked every time a new request is made to the `gevent.server.StreamingServer` you set up below.  That's why the arguments passed to this function are a single connection and the address of the client socket.  Your job here is only to accumulate and then handle the one request that will come over this client socket.